jwst/resample/resample_spec.py

```diff
@@ -60,8 +60,6 @@ def __init__(self, input_models, output=None, single=False, blendheaders=False,
         kwargs : dict
             Other parameters
         """
-        self.input_models = input_models
```

Comment:
The main reason is due to the use of `ResampleData` from outlier detection:
https://github.com/spacetelescope/jwst/blob/bc255f4cffa7a887ab3d1eed7ebfe82b856a4c67/jwst/outlier_detection/outlier_detection.py#L148-L150
Since `ResampleData` was updated and `ResampleSpecData` is a subclass I updated it as well. This pattern is problematic (as described in the PR description) and causes python to hold onto the input data well beyond when it's useful.